src/allmydata/client.py

```diff
@@ -179,42 +203,86 @@ def _sequencer(self):
+                                },
+                            },
+                            }
+            new_connections = connections.copy()
+            new_connections['introducers'][u'default'] = {}
+            new_connections['introducers']['default']['furl'] = furl
+            connections_filepath.setContent(yaml.dump(new_connections))
+        return connections, exists
+
+    def load_connections(self):
+        """
+        Load the connections.yaml file if it exists, otherwise
+        create a default configuration. Abort startup and report
+        an error to the user if the tahoe.cfg contains an introducer
+        FURL which is also found in the connections.yaml.
+        """
+        # read introducer from tahoe.cfg and abort + error an introducer furl is specified
+        # which is also found in our connections.yaml
+        self.introducer_furls = [] # XXX
+        tahoe_cfg_introducer_furl = self.get_config("client", "introducer.furl", None)
         self.warn_flag = False
-        # Try to load ""BASEDIR/private/introducers" cfg file
-        cfg = os.path.join(self.basedir, "private", "introducers")
-        if os.path.exists(cfg):
-            f = open(cfg, 'r')
-            for introducer_furl in f.read().split('\n'):
-                introducer_furl_stripped = introducer_furl.strip()
-                if introducer_furl_stripped.startswith('#') or not introducer_furl_stripped:
-                    continue
-                self.introducer_furls.append(introducer_furl_stripped)
-            f.close()
-        furl_count = len(self.introducer_furls)
-
-        # read furl from tahoe.cfg
-        ifurl = self.get_config("client", "introducer.furl", None)
-        if ifurl and ifurl not in self.introducer_furls:
-            self.introducer_furls.append(ifurl)
-            f = open(cfg, 'a')
-            f.write(ifurl)
-            f.write('\n')
-            f.close()
-            if furl_count > 1:
+
+        connections, connections_yaml_exists = self.load_connections_from_yaml(tahoe_cfg_introducer_furl)
+        introducers = connections['introducers']
+        transports = connections['transport_plugins']
+        if self.tub is None:
+            return
+        plugins = load_plugins(connections['transport_plugins'])
+        self.tub.removeAllConnectionHintHandlers()
+        for name, handler in plugins.items():
+            self.tub.addConnectionHintHandler(name, handler)
+
+        found = False
+        count = 0
+        if tahoe_cfg_introducer_furl is not None and connections_yaml_exists:
+            count += 1
+            for nick in introducers.keys():
+                if tahoe_cfg_introducer_furl == introducers[nick]['furl']:
+                    found = True
+                    break
+            if not found and count > 0:
+                log.err("Introducer furl %s specified in both tahoe.cfg and connections.yaml; please fix impossible configuration.")
```

Comment:
this log line is incorrect (and matches the incorrect comment): the actual error condition, which i believe is correctly implemented (but am not certain due to the obtuse logic) is when the introducer furl is in `tahoe.cfg` but NOT in the yaml (but the yaml already existed).